mos-platform/atari5200-super/link.ld

```diff
@@ -0,0 +1,278 @@
+  __bank14_fixed_size = __cart_rom_size >= 512 ? __fixed_size : 0;
+  __bank15_fixed_size = __cart_rom_size >= 512 ? __fixed_size : 0;
+
+  .bank0 __bank0_lma + __fixed_size : {
+    *(.bank0 .bank0.*)
+  } >bank0
+
+  .bank1 __bank1_lma + __bank1_fixed_size : {
+    *(.bank1 .bank1.*)
+  } >bank1
+
+  .bank2 __bank2_lma + __bank2_fixed_size : {
+    *(.bank2 .bank2.*)
+  } >bank2
+
+  .bank3 __bank3_lma + __bank3_fixed_size : {
+    *(.bank3 .bank3.*)
+  } >bank3
+
+  .bank4 __bank4_lma + __bank4_fixed_size : {
+    *(.bank4 .bank4.*)
+  } >bank4
+
+  .bank5 __bank5_lma + __bank5_fixed_size : {
+    *(.bank5 .bank5.*)
+  } >bank5
+
+  .bank6 __bank6_lma + __bank6_fixed_size : {
+    *(.bank6 .bank6.*)
+  } >bank6
+
+  .bank7 __bank7_lma + __bank7_fixed_size : {
+    *(.bank7 .bank7.*)
+  } >bank7
+
+  .bank8 __bank8_lma + __bank8_fixed_size : {
+    *(.bank8 .bank8.*)
+  } >bank8
+
+  .bank9 __bank9_lma + __bank9_fixed_size : {
+    *(.bank9 .bank9.*)
+  } >bank9
+
+  .bank10 __bank10_lma + __bank10_fixed_size : {
+    *(.bank10 .bank10.*)
+  } >bank10
+
+  .bank11 __bank11_lma + __bank11_fixed_size : {
+    *(.bank11 .bank11.*)
+  } >bank11
+
+  .bank12 __bank12_lma + __bank12_fixed_size : {
+    *(.bank12 .bank12.*)
+  } >bank12
+
+  .bank13 __bank13_lma + __bank13_fixed_size : {
+    *(.bank13 .bank13.*)
+  } >bank13
+
+  .bank14 __bank14_lma + __bank14_fixed_size : {
+    *(.bank14 .bank14.*)
+  } >bank14
+
+  .bank15 __bank15_lma + __bank15_fixed_size : {
+    *(.bank15 .bank15.*)
+  } >bank15
+
+  .fixed     : { *(.fixed     .fixed.*)     } >fixed
+
+  .bank0     : { *(.bank0     .bank0.*)     } >bank0
+  .bank1     : { *(.bank1     .bank1.*)     } >bank1
+  .bank2     : { *(.bank2     .bank2.*)     } >bank2
+  .bank3     : { *(.bank3     .bank3.*)     } >bank3
+  .bank4     : { *(.bank4     .bank4.*)     } >bank4
+  .bank5     : { *(.bank5     .bank5.*)     } >bank5
+  .bank6     : { *(.bank6     .bank6.*)     } >bank6
+  .bank7     : { *(.bank7     .bank7.*)     } >bank7
+  .bank8     : { *(.bank8     .bank8.*)     } >bank8
+  .bank9     : { *(.bank9     .bank9.*)     } >bank9
+  .bank10    : { *(.bank10    .bank10.*)    } >bank10
+  .bank11    : { *(.bank11    .bank11.*)    } >bank11
+  .bank12    : { *(.bank12    .bank12.*)    } >bank12
+  .bank13    : { *(.bank13    .bank13.*)    } >bank13
+  .bank14    : { *(.bank14    .bank14.*)    } >bank14
+  .bank15    : { *(.bank15    .bank15.*)    } >bank15
+
+
+  /* Tail: 64 bytes in total
+
+     bank-switch-read-addresses, 16 bytes; from the platform PoV there
+     are from 0xbfc0 through 0xbfcf
+
+     ... 24 unused bytes ...
+
+     Cartridge Metadata: 24 bytes
+
+     20 bytes of title at bfe8 (ATASCII)
+     2 year 19xx at bffc - if bffd is 0xff this means diag cart - TODO
+     init at bffe
+
+     TODO: We might want to juse use things in diag cart mode (some
+     games do this) and ignore the above.  */
+  .vector __tail_lma + (64 - 24) : {
```

Comment:
Could we make these configurable symbols with PROVIDEed values? This is used to great effect in the NES setup.

Seems like this would work for everything except the title; if we wanted that to be configurable (which seems desirable), we'd probably need to make it a weak symbol in a retained section. The user could override the symbol and provide their own 20 byte section to replace the one available in atari2500-super's libcrt.

We should probably do the low hanging fruit as part of this PR (any numeric metadata), but NAR on the ASCII; that would require some experimentation.

Reply:
@mysterymath How allowing a quirky replacement as I have it now?

I agree some mechanism to see the string would be useful, but it also might make sense later on to replace the init code entirely with something we provide, in which case we can do whatever we want in-terms of boot logo, year (at the cost of code-size).